### PR TITLE
Tar study and push to S3 on push GitHub action

### PR DESCRIPTION
**Pull request by mandawilson**
We have added a GitHub workflow that will be triggered on any push to the master branch that is in the cBioPortal repository, that involves a change to the ./public/ directory.

Any study that has changed will be checked out of git, compressed, and copied to an S3 bucket.

Co-authored-by: Avery Wang 18199796+averyniceday@users.noreply.github.com  
Co-authored-by: Robert Sheridan 7747489+sheridancbio@users.noreply.github.com

### CODE REVIEW

**Comment by mandawilson** [comment]
Don't forget to compress the commits!

File: .github/workflows/zip_studies.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          git lfs pull --include="${changed_study}"
+          echo "Compressing this study: ${changed_study}."
+          tar -czvf ../studies_to_upload/${changed_study}.tar.gz ${changed_study}
+          echo "Copy file to S3: ../studies_to_upload/${changed_study}.tar.gz"
+          aws s3 cp --acl public-read ../studies_to_upload/${changed_study}.tar.gz s3://${AWS_S3_BUCKET}/
+          echo "Remove this directory: ${changed_study}."
+          rm -rf "${changed_study}"
+          echo "Remove this file: ../studies_to_upload/${changed_study}.tar.gz."
+          rm ../studies_to_upload/${changed_study}.tar.gz
+          echo "Remove rm -rf .git"
```
Comment (sheridancbio) [inline .github/workflows/zip_studies.yml:50]: should probably just say "Remove .git"

File: .github/workflows/zip_studies.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      - 'public/**'
+jobs:
+  zip_studies:
+    if: github.repository == 'cbioportal/datahub'
+    runs-on: ubuntu-latest
+    timeout-minutes: 1440
+    steps:
+    - id: files
+      uses: jitterbit/get-changed-files@v1
+    - run: | # TODO remove step
```
Comment (sheridancbio) [inline .github/workflows/zip_studies.yml:18]: This, and the next few steps should either be removed, or the TODO comment should be removed.

File: .github/workflows/zip_studies.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        exit 0
+    - run: |
+        echo "##[set-output name=unique_studies_list;]$(echo "${{ steps.files.outputs.all }}" | awk 'BEGIN { RS=" "; FS="/"; ORS=" " } { set[$2]++ } END { for (dir in set) { print dir } }')"
+        exit 0
+      id: unique_studies
+    - run: | # TODO remove step
+        echo ${{ steps.unique_studies.outputs.unique_studies_list }}
+        exit 0
+    - run: |
+        git --version
```
Comment (sheridancbio) [inline .github/workflows/zip_studies.yml:29]: This probably can be removed too

File: .github/workflows/zip_studies.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        exit 0
+    - uses: actions/checkout@v2
+    #  with:
```
Comment (sheridancbio) [inline .github/workflows/zip_studies.yml:32]: maybe remove these two commented lines? Or, put them in with "lfs: 'false'" if that is appropriate.

File: .github/workflows/zip_studies.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+          echo "Remove rm -rf .git"
+          cd ..
+          rm -rf .git
+          # add original .git back
+          tar -xf savegit.tar
+          cd public
+        done
+        cd ..
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (sheridancbio) [inline .github/workflows/zip_studies.yml:59]: I suppose we could have the savegit.tar filename as an environment variable declared here?

**Review by sheridancbio — APPROVED** [review]
Looks good .. but maybe remove some comments

**Pull request merged**